include/s3select.h

```diff
@@ -2133,48 +2143,89 @@ class base_s3object
   }
 
   void result_values_to_string(multi_values& projections_resuls, std::string& result)
-  {
-    size_t i = 0;
+{   
+    std::string last_key;
     std::string output_delimiter(1,m_csv_defintion.output_column_delimiter);
     std::string output_row_delimiter(1,m_csv_defintion.output_row_delimiter);
+    std::vector<std::string> key_path(m_sa->get_json_key_path());
+    
+    if(key_path.size()) {
+      last_key = key_path[key_path.size()-1];
+    }
 
-    for(auto& res : projections_resuls.values)
-    {
-	    if(fp_ext_debug_mesg)
-		      fp_ext_debug_mesg( res->to_string() );
+    if(is_json)  {
 
-            if (m_csv_defintion.quote_fields_always) {
+      result += "{";
+      for (size_t i = 0; i < projections_resuls.values.size(); ++i)
+      {
+          auto& res = projections_resuls.values[i];
+          std::string label = "_";
+          label += std::to_string(i + 1);
+
+          if (i > 0) {
+              result += output_delimiter;
+          }
+          
+          if(!is_star && m_aggr_flow)  {
+            result += "\"" + label + "\":";
```

Comment:
this labeling format procedure is not just for aggregation flow, it is also for any function operators
such as :
_1.my_key + 1
non_aggregate_function( _1.my_key )